### ISSUE TITLE
Slider changes value in bean even when user did not changed it

### ISSUE THREAD

**Issue by SomMeri** [vaadin/framework#11890]
#### Versions and Systems
Vaadin Framework version: 8.10.0 (It works correctly on older versions)
Browser version: all of them
Web container name and version: all of them

#### Description of the bug
When I set bean to slider, the slider rounds the value inside the bean. This happens without any user interaction. The bean value should change only if the user moves the slider, the value should not be modified by the `binder.setBean(bean)` call.

#### Minimal reproducible example
````java
Slider slider = new Slider();
slider.setResolution(2);

Label label = new Label();

SliderBean bean = new SliderBean();
bean.setValue(100.003);


System.out.println("bean value before: " + bean.getValue());

Binder<SliderBean> binder = new Binder<>();
binder.forField(slider).bind(SliderBean::getValue, SliderBean::setValue);
binder.setBean(bean);
System.out.println("bean value after: " + bean.getValue());
````
#### Expected behavior
Bean value remains 100.003 until user moves slider. Console contains:
````
bean value before: 100.003
bean value after: 100.003
````

#### Actual behavior
Bean value is changed to 100.0. Console contains:
````
bean value before: 100.003
bean value after: 100.0
````

#### Further Analysis
This is new problem in 8.10.0, older version does not have the issue. 8.9.4 works correctly.

The issue is likely caused by these [new lines](https://github.com/vaadin/framework/blob/e5118c796f1eabe6f05c3f8773af5c296f06e849/server/src/main/java/com/vaadin/data/Binder.java#L1203-L1208) in `initFieldValue` method. The `setBean` method calls if [from here](https://github.com/vaadin/framework/blob/e5118c796f1eabe6f05c3f8773af5c296f06e849/server/src/main/java/com/vaadin/data/Binder.java#L1761).

**Comment by TatuLund** [comment]
Your analysis about what is the cause of the change of behavior here is correct.

We did already while ago this change, which addresses some issues like this one

https://github.com/vaadin/framework/issues/9000

Also it is worth of mentioning, that as it was slightly behavior altering change the fix was postponed to Vaadin 8.10. The same fix is also in our Vaadin 14 product, so now Binder behavior is the same in both, and this change in Vaadin 8 makes it forward compatible.

So this is not a bug, but intentional behavior change.

I would say that also in this particular case the behavior is now correct.

**Comment by SomMeri** [comment]
@TatuLund Could it be made easily customizable? The tricks I have to currently do to keep original value in bean are ugly and fragile.

Workaround in case someone else needs this too:
````java 

public class InputFieldBinder<M> extends Binder<M> {
        [...]

	@Override
	protected <FIELDVALUE, TARGET> BindingBuilder<M, TARGET> doCreateBinding(HasValue<FIELDVALUE> field, Converter<FIELDVALUE, TARGET> converter, BindingValidationStatusHandler handler) {
		return new KeepInitialValueBindingBuilderImpl<>(this, field, converter, handler);
	}

	/**
	 * See {@link KeepInitialValueBindingBuilderImpl} to see why this is done.
	 */
	private boolean ignoreValueChange = false;

	@Override
	public void setBean(M bean) {
		try {
			ignoreValueChange = true;
			super.setBean(bean);
		} finally {
			ignoreValueChange = false;
		}
	}

	/**
	 * Since vaadin 8.10.0, vaadin changes bean value upon setBean method. This primarily affected the slider -
	 * it rounded value inside the bean. 
	 * 
	 * This is workaround. We do not want to change stored value unless user modified it.
	 * 
	 *
	 * @param <FIELDVALUE>
	 * @param <TARGET>
	 */
	class KeepInitialValueBindingBuilderImpl<FIELDVALUE, TARGET> extends BindingBuilderImpl<M, FIELDVALUE, TARGET> {

		protected KeepInitialValueBindingBuilderImpl(Binder<M> binder,
				HasValue<FIELDVALUE> field,
				Converter<FIELDVALUE, TARGET> converterValidatorChain,
				BindingValidationStatusHandler statusHandler) {
			super(binder, field, converterValidatorChain, statusHandler);
		}

		@Override
		public Binding<M, TARGET> bind(ValueProvider<M, TARGET> getter, Setter<M, TARGET> setter) {
			return super.bind(getter, (bean, value) -> hackedSetter(bean, value, setter));
		}

		private void hackedSetter(M bean, TARGET value, Setter<M, TARGET> setter) {
			if (!ignoreValueChange) {
				setter.accept(bean, value);
			}
		}

	}
}
````
